### PR TITLE
test: Staking module e2e tests

### PR DESCRIPTION
**Pull request by glnro**
This PR fulfills basic test commitment for #1630 and introduces a new suite of tests for the staking module.

CLI commands tested
- `keys`
  - `add` Add new key
- `tx staking`
  - `delegate` Delegate staked tokens to a validator
  - `redelegate` Redelegate previously staked tokens to a new validator

REST endpoints tested
- [Delegation](https://docs.cosmos.network/v0.46/modules/staking/09_client.html#delegation-3) `/cosmos/staking/v1beta1/validators/{validatorAddresess}/delegations/{delegatorAddress}` 

User Flows
- A user wants to delegate tokens to a known validator
- A user wants to redelegate tokens that are staked to a new validator

This PR also includes a few refactors related to minor formatting in the group tests.

### CODE REVIEW

File: tests/e2e/e2e_cli_util.go [hunk 1]
```diff
@@ -104,24 +105,23 @@ func queryGaiaDenomBalance(endpoint, addr, denom string) (sdk.Coin, error) {
...
 func queryGovProposal(endpoint string, proposalID int) (govv1beta1.QueryProposalResponse, error) {
-	var emptyProp govv1beta1.QueryProposalResponse
+	var govProposalResp govv1beta1.QueryProposalResponse
 
 	path := fmt.Sprintf("%s/cosmos/gov/v1beta1/proposals/%d", endpoint, proposalID)
 
 	resp, err := http.Get(path) //nolint:gosec // this is only used during tests
 	if err != nil {
-		return emptyProp, fmt.Errorf("failed to execute HTTP request: %w", err)
+		return govProposalResp, fmt.Errorf("failed to execute HTTP request: %w", err)
```
Comment (mmulji-ic) [inline tests/e2e/e2e_cli_util.go:114]: Would've preferred to keep the group stuff in a separate PR

File: tests/e2e/e2e_setup_test.go [hunk 2]
```diff
@@ -55,7 +55,7 @@ const (
 
 var (
 	stakeAmount       = math.NewInt(100000000000)
-	stakeAmountCoin   = sdk.NewCoin("stake", stakeAmount)
+	stakeAmountCoin   = sdk.NewCoin(uatomDenom, stakeAmount)
```
Comment (mmulji-ic) [inline tests/e2e/e2e_setup_test.go]: Mismatch in denoms? uatom/stake?

File: tests/e2e/e2e_groups_test.go [hunk 3]
```diff
@@ -54,33 +52,30 @@ var (
 )
 
 func (s *IntegrationTestSuite) GroupsSendMsgTest() {
-	ctx, cancel := context.WithTimeout(context.Background(), time.Minute)
-	defer cancel()
-
 	chainAAPIEndpoint := fmt.Sprintf("http://%s", s.valResources[s.chainA.id][0].GetHostPort("1317/tcp"))
-	s.setup(ctx)
+	s.setup()
```
Comment (mmulji-ic) [inline tests/e2e/e2e_groups_test.go:56]: Does .setup() default to without timeout?

File: tests/e2e/e2e_cli_util.go (hunk 1 above)
Comment (glnro) [reply tests/e2e/e2e_cli_util.go:114]: Yeah started to get ahead of myself, I'll follow up with a more thorough refactor pr

File: tests/e2e/e2e_setup_test.go (hunk 2 above)
Comment (glnro) [reply tests/e2e/e2e_setup_test.go]: I changed the bonding denom to uatom to keep everything in the test consistent...tweaked the variable names to reflect it's the amount for staking not a stake denom amount.

File: tests/e2e/e2e_cli_util.go [hunk 4]
```diff
@@ -3,6 +3,7 @@ package e2e
 import (
 	"encoding/json"
 	"fmt"
+	staketypes "github.com/cosmos/cosmos-sdk/x/staking/types"
 	"io"
```
Comment (yaruwangway) [inline tests/e2e/e2e_cli_util.go]: formatting:
staketypes "github.com/cosmos/cosmos-sdk/x/staking/types" is at last with line space.

File: tests/e2e/genesis.go [hunk 5]
```diff
@@ -118,13 +119,21 @@ func modifyGenesis(path, moniker, amountStr string, accAddr sdk.AccAddress, glob
...
+	}
+	appState[staketypes.ModuleName] = stakingGenStateBz
+
 	// Refactor to separate method
 	amnt := math.NewInt(10000)
 	quorum, _ := sdk.NewDecFromStr("0.000000000000000001")
 	threshold, _ := sdk.NewDecFromStr("0.000000000000000001")
 
 	govState := govv1beta1.NewGenesisState(1,
-		govv1beta1.NewDepositParams(sdk.NewCoins(sdk.NewCoin("uatom", amnt)), 10*time.Minute),
```
Comment (yaruwangway) [inline tests/e2e/genesis.go:127]: why pass denom as argument. since amt is hard-coded here, `amnt := math.NewInt(10000)` can just hard code denom

File: tests/e2e/e2e_staking_test.go [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+	validatorB := s.chainA.validators[1]
+	sender, err := validatorA.keyInfo.GetAddress()
+	s.NoError(err)
+	validatorBAddr, err := validatorB.keyInfo.GetAddress()
+	s.NoError(err)
+
+	valOperA := sdk.ValAddress(sender)
+	valOperB := sdk.ValAddress(validatorBAddr)
+
+	alice := s.executeGKeysAddCommand(s.chainA, 0, "alice", home)
```
Comment (yaruwangway) [inline tests/e2e/e2e_staking_test.go:27]: we can refactor here later, i added an addr to genesis as ica owner, who  already has some tokens. so that we can simplify here that no need to add keys, and fund alice, can use ica onwer as delegator. but this solution is also good!

File: tests/e2e/genesis.go [hunk 7]
```diff
@@ -3,6 +3,7 @@ package e2e
 import (
 	"encoding/json"
 	"fmt"
+	staketypes "github.com/cosmos/cosmos-sdk/x/staking/types"
```
Comment (yaruwangway) [inline tests/e2e/genesis.go]: formatting

File: tests/e2e/e2e_groups_test.go [hunk 8]
```diff
@@ -246,8 +228,8 @@ func (s *IntegrationTestSuite) setup(ctx context.Context) {
 	s.Require().NoError(err)
 	aliceAddr = alice.String()
 
-	bobAddr = s.executeGaiaKeysAddCommand(ctx, s.chainA, 0, "bob")
-	charlieAddr = s.executeGaiaKeysAddCommand(ctx, s.chainA, 0, "charlie")
+	bobAddr = s.executeGKeysAddCommand(s.chainA, 0, "bob", "/home/nonroot/.gaia")
```
Comment (mmulji-ic) [inline tests/e2e/e2e_groups_test.go:231]: Hard coding directory, when above in dataDirectoryHome you avoided.

File: tests/e2e/e2e_util_test.go [hunk 9]
```diff
@@ -863,3 +878,103 @@ func (s *IntegrationTestSuite) executeGaiaKeysAddCommand(ctx context.Context, c
...
+func (s *IntegrationTestSuite) executeKeysList(c *chain, valIdx int, home string) {
+	ctx, cancel := context.WithTimeout(context.Background(), time.Minute)
+	defer cancel()
+
+	var (
+		outBuf bytes.Buffer
+		errBuf bytes.Buffer
+	)
+
+	gaiaCommand := []string{
```
Comment (mmulji-ic) [inline tests/e2e/e2e_util_test.go:891]: This kind of operation I've isolated in a helper fn, seems unnecessary to repeat, why not adapt executeGaiaTxCommand

Comment (glnro) [reply tests/e2e/e2e_util_test.go:891]: Yeah so there's the main helper function `executeGaiaTxCommand` and the difference in this case is it queries the transaction hash after executing to confirm the transaction didn't fail...we could refactor this further to a general purpose function that know whether it's executing a tx or keys command but I fear refactoring too much at this point, lmk what you think if we should go ahead and just do this

File: tests/e2e/genesis.go (hunk 5 above)
Comment (glnro) [reply tests/e2e/genesis.go:127]: either or I just thought it would be cleaner to refactor to using the denom const everywhere as we go

File: tests/e2e/e2e_staking_test.go (hunk 6 above)
Comment (glnro) [reply tests/e2e/e2e_staking_test.go:27]: Sure, I'll refactor to this way in the `x/dist` test
